util/optional_mutex.h

```diff
@@ -0,0 +1,20 @@
+#ifndef OPTIONAL_MUTEX_H_
+#define OPTIONAL_MUTEX_H_
+
+#include <pthread.h>
+
+/**
+ * optional_mutex_lock(mutex):
+ * If OPTIONAL_MUTEX_PTHREAD_YES is defined, call pthread_mutex_lock();
+ * if OPTIONAL_MUTEX_PTHREAD_NO is defined, do nothing.
+ */
+int optional_mutex_lock(pthread_mutex_t *);
```

Comment:
I considered making this take `void *` instead.  That would eliminate the requirement to `#include <pthread.h>` in the header, which in turn would be an odd thing to be present if we're not doing `-lpthread`.

In the end I went with the clarity of using `pthread_mutex_t *` rather than `void *` as the arguments, but it was a 55%-45% decision.